applications/vanilla/models/class.categorymodel.php

```diff
@@ -1035,6 +1039,155 @@ private function gatherLastIDs($categoryTree, &$result = null) {
     }
 
     /**
+     * Given a discussion, update its category's last post info and counts.
+     *
+     * @param int|array|stdClass $discussion The discussion ID or discussion.
+     */
+    public function incrementLastDiscussion($discussion) {
+        // Lookup the discussion record, if necessary. We need at least a discussion to continue.
+        if (is_numeric($discussion)) {
```

Comment:
`is_numeric(9.1) === true`
`is_numeric('0xf4c3b00c') === true` prior to php7.0

Use `filter_var($value, FILTER_VALIDATE_INT) !== false` or `ctype_digit((string)$value)`